### ISSUE TITLE
Using attribute-from-tuple conversion can be error-prone

### ISSUE THREAD

**Issue by dralley** [tafia/quick-xml#410]
Creating an attribute from a `(&[u8], &[u8])` performs no transformation (escaping) on the provided value, but creating an attribute from `(&str, &str)` does.

This feels a bit too implicit.  It should probably be treated like `BytesText` instead, where different methods are provided depending on whether the value has been pre-escaped.  https://docs.rs/quick-xml/latest/quick_xml/events/struct.BytesText.html

```Rust
use quick_xml::events::attributes::Attribute;

fn main() {
    // correct
    println!("{:#?}", Attribute::from(("foo", "Bells & Whistles")));
    println!("{:#?}", Attribute::from(("foo".as_bytes(), "Bells &amp; Whistles".as_bytes())));

    // incorrect
    println!("{:#?}", Attribute::from(("foo", "Bells &amp; Whistles")));
    println!("{:#?}", Attribute::from(("foo".as_bytes(), "Bells & Whistles".as_bytes())));
}
```

```
Attribute { key: "foo", value: Owned("Bells &amp; Whistles") }
Attribute { key: "foo", value: Borrowed("Bells &amp; Whistles") }
Attribute { key: "foo", value: Owned("Bells &amp;amp; Whistles") }
Attribute { key: "foo", value: Borrowed("Bells & Whistles") }
```

**Comment by Mingun** [comment]
I think, we should leave only `From<(&str, &str)>` (with current behavior) and replace `From<(&[u8], &[u8])>` with an explicit `new` method

**Comment by dralley** [comment]
That seems reasonable, but do you see a reason not to mirror the `BytesText` API here?

e.g.

* `Attribute::from_escaped_bytes`
* `Attribute::from_bytes`
* `Attribute::from_escaped_str`
* `Attribute::from_str`

(I don't love the `_plain_` naming, admittedly)

**Comment by Mingun** [comment]
Yes, I think, we should stick to consistent naming. That methods seems reasonable. I think, that we could leave `From<(&str, &str)>` anyway because it seems ergonomic

**Comment by dralley** [comment]
The `from((&[u8], &[u8])) ` constructor no longer exists, thus the crux of this issue no longer exists.
